routers/repo/view.go

```diff
@@ -252,7 +252,7 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		}
 
 		if ctx.Repo.CanEnableEditor() {
-			ctx.Data["CanEditFile"] = true
+			ctx.Data["CanEditFile"] = !ctx.Repo.Repository.IsArchived
```

Comment:
Why not just change `ctx.Repo.CanEnableEditor` ?

Reply:
Right, fixed.